src/p2p/behaviour.rs

```diff
@@ -246,7 +384,16 @@ impl<Types: IpfsTypes> Behaviour<Types> {
         }
     }
 
-    pub fn add_peer(&mut self, peer: PeerId) {
+    pub fn add_peer(&mut self, peer: PeerId, addr: Multiaddr) {
+        self.kademlia.add_address(&peer, addr);
+        /*
+                match self.kademlia.start_providing(peer.to_base58().into_bytes().into()) {
+                    Ok(id) => {
+                        self.kad_queries.insert(id);
+                    },
+                    Err(e) => error!("kad: can't provide peer {}: {:?}", peer, e),
+                }
+        */
```

Comment:
This commented out code seems a bit odd. I would had expected the routing tables would be used without explicitly providing anything for a key. Would something be stored under the other peers key?

Reply:
this should be removed, I forgot to do that; I initially didn't know if it was a necessary step